deltachat-ffi/src/lib.rs

```diff
@@ -1144,7 +1127,41 @@ pub unsafe extern "C" fn dc_delete_chat(context: *mut dc_context_t, chat_id: u32
             .delete(&ctx)
             .await
             .log_err(ctx, "Failed chat delete")
-            .unwrap_or(())
+            .ok();
+    })
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn dc_block_chat(context: *mut dc_context_t, chat_id: u32) {
+    if context.is_null() {
+        eprintln!("ignoring careless call to dc_block_chat()");
+        return;
+    }
+    let ctx = &*context;
+
+    block_on(async move {
+        ChatId::new(chat_id)
+            .block(&ctx)
+            .await
+            .log_err(ctx, "Failed chat block")
+            .ok();
+    })
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn dc_unblock_chat(context: *mut dc_context_t, chat_id: u32) {
```

Comment:
possibly same issue as `dc_block_chat()`

Reply:
Same, I copied `dc_delete_chat` interface. Looks like C interface does not return errors in most cases, as there is nothing UI can do about it anyway. UI requests the core to do something, then the core sends events if something happens (error or chat changed)